src/Simplex/Chat.hs

```diff
@@ -174,49 +174,87 @@ processChatCommand = \case
     CTGroup -> CRApiChat . AChat SCTGroup <$> withStore (\st -> getGroupChat st user cId pagination)
     CTContactRequest -> pure $ chatCmdError "not implemented"
   APIGetChatItems _pagination -> pure $ chatCmdError "not implemented"
-  APISendMessage cType chatId _file mc -> withUser $ \user@User {userId} -> withChatLock $ case cType of
-    -- TODO send message with file attachment; initiate file transfer
+  APISendMessage cType chatId file_ quotedItemId_ mc -> withUser $ \user@User {userId} -> withChatLock $ case cType of
     CTDirect -> do
-      ct <- withStore $ \st -> getContact st userId chatId
-      sendNewMsg user ct (MCSimple (ExtMsgContent mc Nothing)) mc Nothing
-    CTGroup -> do
-      group@(Group GroupInfo {membership} _) <- withStore $ \st -> getGroup st user chatId
-      unless (memberActive membership) $ throwChatError CEGroupMemberUserRemoved
-      sendNewGroupMsg user group (MCSimple (ExtMsgContent mc Nothing)) mc Nothing
-    CTContactRequest -> pure $ chatCmdError "not supported"
-  APISendMessageQuote cType chatId quotedItemId _file mc -> withUser $ \user@User {userId} -> withChatLock $ case cType of
-    -- TODO send message with file attachment; initiate file transfer
-    CTDirect -> do
-      (ct, qci) <- withStore $ \st -> (,) <$> getContact st userId chatId <*> getDirectChatItem st userId chatId quotedItemId
-      case qci of
-        CChatItem _ ChatItem {meta = CIMeta {itemTs, itemSharedMsgId}, content = ciContent, formattedText} -> do
-          case ciContent of
-            CISndMsgContent qmc -> send_ CIQDirectSnd True qmc
-            CIRcvMsgContent qmc -> send_ CIQDirectRcv False qmc
-            _ -> throwChatError CEInvalidQuote
+      ct@Contact {localDisplayName = c} <- withStore $ \st -> getContact st userId chatId
+      (fileInvitation_, ciFile_) <- unzipMaybe <$> setupSndFileTransfer ct
+      (msgContainer, quotedItem_) <- prepareMsg fileInvitation_
+      msg <- sendDirectContactMessage ct (XMsgNew msgContainer)
+      ci <- saveSndChatItem user (CDDirectSnd ct) msg (CISndMsgContent mc) ciFile_ quotedItem_
```

Comment:
sendDirectContactMessage + saveSndChatItem are used together in quite a few places, can't it be just one function?